keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/ResourceControllerTests.kt

```diff
@@ -55,9 +58,14 @@ internal class ResourceControllerTests {
     resourceRepository.dropAll()
   }
 
+  @BeforeEach
+  fun mockAuthz() {
+    every { authorizationSupport.userCan(Action.READ.name, Entity.RESOURCE.name, any()) } returns true
+  }
+
   @Test
   fun `can't diff a resource when unauthorized`() {
-    every { authorizationSupport.userCanModifySpec("fzlem@netflix.com", any()) } returns false
+    every { authorizationSupport.userCan(Action.READ.name, Entity.RESOURCE.name, any()) } returns false
```

Comment:
This is more or less what I was trying to accomplish in the minutest-version of controller tests. It works fine here just running with JUnit... 🤷‍♂

Reply:
You know you can nest `context` blocks, right? You could have a structure like:

```kotlin
context("API auth checks") {
  context("user is not authorized") {
    before {
      // set up mock to return false
    }
    test("user can't do the thing") {
      // ...
    }
  }
  context("user is totally authorized") {
    before {
      // set up mock to return true
    }
    test("user is allowed to do the thing") {
      // ...
    }
  }
}
```

Am I missing something?

Reply:
No, and I've tried all sorts of combinations like that. I suspect the issue is when you define a `before` block directly under `tests` (i.e. scoped to all `context`s and `test`s underneath).

Reply:
But why would you initialize the mock to return either true or false there?

Reply:
Well, I had 4 contexts in the class, and in only 1 of each I wanted the behavior to be different. Is the common setup to repeat the same code in all 3 contexts where the behavior is the same? That's what was a bit surprising to me.

Reply:
so have

context 1
  context 1a
  context 1b
  context 1c
context 2